lib/src/bluetooth_device.dart

```diff
@@ -128,8 +128,16 @@ class BluetoothDevice {
       ..remoteId = id.toString()
       ..mtu = desiredMtu;
 
-    return FlutterBlue.instance._channel
+    await FlutterBlue.instance._channel
```

Comment:
as I am new to dart I am not sure if it is true what I write, but is this really making any difference?

you do at the moment within the async method:

```dart
await FlutterBlue.instance._channel.invokeMethod('...');
await FlutterBlue.instance._methodStream.where('.. MtuSize...')
```
while as the old version was returning a future you could have done from the outside
```dart
await btDevice.requestMtu(111); // which is like your first await within the `requestMtu()` method
await btDevice.mtu.first; // which is like your second await within the `requestMtu()` method
```

so I might be wrong (seriously!) but I don't see that this change actually changes anything

Reply:
Yes, you are right, but that code is incomplete, I've sent a complete PR right after the first, and you absolutely need the await keyword in that code.

https://github.com/pauldemarco/flutter_blue/pull/579#issuecomment-628325875

Reply:
Now I see that i've missed a return in last code, you can add it, but it is working for me as is.

Reply:
why don't you update your pull request then? I was (and still am) a little bit confused about a pull request which has updates in comments :thinking:

Reply:
besides that, I still don't get it then (which might be, as I am still confused about darts async style). I thought the
```dart
Future<void> requestMtu(int desiredMtu) async {
   // var stuff..

   return FlutterBlue.instance._channel.invokeMethod('requestMtu', request.writeToBuffer());
}
```
and then doing in another
```dart
Future<void> someMethod() async {
  await this.btDevice.requestMtu(123);
}
```
is the same as doing it simply in your method.. and the second part you can do as well in your method.. I still don't see it or the async stuff is not working (for me) as I would it expect to be.

Reply:
I don't know how pull request works in github, I've never used that, I've posted my code (and repairs I've done) to help other users to benefit from that, but I couldn't change my original (and useless) PR so I left a response below that. 
I thought that what I've done was enough for the author review and integrate that code into the repo. 

As for the second question, there is nothing weird about ```await``` or ```async```, the fact is that the function does not end in:

```dart
return FlutterBlue.instance._channel.invokeMethod('requestMtu', request.writeToBuffer());
``` 

Invoking ```requestMtu``` will call this function inside android, then it will wait a confirmation event. So you must request the MTU and wait for the stream channel to return when the MTU has changed, if you don't add this code it won't work:

```dart
    await FlutterBlue.instance._channel.invokeMethod('requestMtu', 
      request.writeToBuffer());

    await FlutterBlue.instance._methodStream
      .where((m) => m.method == "MtuSize")
      .map((m) => m.arguments)
      .map((buffer) => protos.MtuSizeResponse.fromBuffer(buffer))
      .where((p) => p.remoteId == id.toString())
      .map((p) => p.mtu)
      .first;
```

I've copied and adapted that code from another place inside the sources, so I asked to validate the code before commiting any changes.

Finally, you can use promises of course:
```dart
    return FlutterBlue.instance._channel.invokeMethod('requestMtu', request.writeToBuffer()).then(
      (_) => FlutterBlue
        .instance._methodStream
        .where((m) => m.method == "MtuSize")
        .map((m) => m.arguments)
        .map((buffer) => protos.MtuSizeResponse.fromBuffer(buffer))
        .where((p) => p.remoteId == id.toString())
        .map((p) => p.mtu)
        .first);
```

and await outside:
```dart
Future<void> someMethod() async {
  await this.btDevice.requestMtu(123);
}
```

Reply:
I think I get the issue now and it is mainly not really in this code per se.. your solution is one way. The other would be to fix
```dart
  /// The MTU size in bytes
  Stream<int> get mtu async* {
    yield await FlutterBlue.instance._channel
        .invokeMethod('mtu', id.toString())
        .then((buffer) => new protos.MtuSizeResponse.fromBuffer(buffer))
        .then((p) => p.mtu);

    yield* FlutterBlue.instance._methodStream
        .where((m) => m.method == "MtuSize")
        .map((m) => m.arguments)
        .map((buffer) => new protos.MtuSizeResponse.fromBuffer(buffer))
        .where((p) => p.remoteId == id.toString())
        .map((p) => p.mtu);
  }
```

as this one returns the "right now MTU" initially.. that's why you cannot wait properly for the change.. removing the first `yield await` would make it possible to simply do

```dart
await device.requestMtu(247);
await device.mtu.first;
```